src/linear_algebra/matrix/block.lean

```diff
@@ -209,4 +259,104 @@ lemma det_of_lower_triangular [linear_order m] (M : matrix m m R) (h : M.block_t
   M.det = ∏ i : m, M i i :=
 by { rw ←det_transpose, exact det_of_upper_triangular h.transpose }
 
+/-! ### Invertible -/
+
+lemma to_block_inverse_mul_to_block_eq_one_of_block_triangular [linear_order α]
+  [invertible M] (hM : block_triangular M b) (k : α) :
+  M⁻¹.to_block (λ i, b i < k) (λ i, b i < k) ⬝ M.to_block (λ i, b i < k) (λ i, b i < k) = 1 :=
+begin
+  let p := (λ i, b i < k),
+  have h_sum : M⁻¹.to_block p p ⬝ M.to_block p p +
+      M⁻¹.to_block p (λ i, ¬ p i) ⬝ M.to_block (λ i, ¬ p i) p = 1,
+    by rw [←to_block_mul_eq_add, inv_mul_of_invertible M, to_block_one_eq],
+  have h_zero : M.to_block (λ i, ¬ p i) p = 0,
+  { ext i j,
+    simpa using hM (lt_of_lt_of_le j.2 (le_of_not_lt i.2)) },
+  simpa [h_zero] using h_sum
+end
+
+/-- The inverse of an upper-left subblock of a block-triangular matrix `M` is the upper-left
+subblock of `M⁻¹`. -/
+lemma inv_to_block_of_block_triangular [linear_order α]
+  [invertible M] (hM : block_triangular M b) (k : α) :
+  (M.to_block (λ i, b i < k) (λ i, b i < k))⁻¹ = M⁻¹.to_block (λ i, b i < k) (λ i, b i < k) :=
+inv_eq_left_inv (to_block_inverse_mul_to_block_eq_one_of_block_triangular hM k)
+
+/-- An upper-left subblock of an invertible block-triangular matrix is invertible. -/
+noncomputable def invertible_to_block_of_block_triangular
```

Comment:
What makes this noncomputable?

Reply:
It used `M⁻¹` instead of `⅟M`. I've made it computable now.